tests/console/avocado_prepare.pm

```diff
@@ -0,0 +1,116 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Avocado setup and network bridge configuration
+#          Test system has to be registered baremetal
+# Maintainer: Jozef Pupava <jpupava@suse.cz>
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use utils qw(systemctl zypper_call);
+use version_utils qw(is_sle is_opensuse);
+use Utils::Backends 'use_ssh_serial_console';
+
+sub teardown {
+    my $teardown = <<'EOF';
+#!/bin/bash
+if [[ $(grep 1 /sys/class/net/br0/carrier) ]]; then
+    ACTIVE_NET=$(grep -l 1 /sys/class/net/eth*/carrier|head -n1|awk -F/ '{print$5}')
+    mv /etc/sysconfig/network/old_ifcfg-$ACTIVE_NET /etc/sysconfig/network/ifcfg-$ACTIVE_NET
+    rm /etc/sysconfig/network/ifcfg-br0
+    systemctl restart network;
+fi
+EOF
+    script_output($teardown);
+    assert_script_run 'ip a';
+}
+
+sub run {
+    use_ssh_serial_console;
+
+    # needed for script_output
+    zypper_call 'in curl';
+
+    my $distribution = is_opensuse() ? 'openSUSE_Tumbleweed' : is_sle('=12-sp4') ? 'SLE_12_SP4' : 'SLE_15';
+    my $python_version = is_opensuse() || is_sle('15+') ? 3 : 2;
+    zypper_call "ar -f -G http://download.opensuse.org/repositories/devel:/languages:/python:/avocado/$distribution/devel:languages:python:avocado.repo", exitcode => [0, 4, 104];
```

Comment:
What about  test suite variable AVOCADO_REPO? In similar way like piglit test has.

Reply:
changed